### PR TITLE
METEOR metrics doc

### PR DESCRIPTION
**Pull request by billbai0102**
### Linked issue(s):
Fixes KOL-3435

### What change does this PR introduce and why?
Adds the METEOR metric doc to the glossary

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: docs/metrics/meteor.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  exclude: true
+---
+
+# METEOR
+
+METEOR (**M**etric for **E**valuation of **T**ranslation with **E**xplicit **OR**dering) is a widely recognized and vital metric used in natural language processing. Originally developed for machine translation workflows, it is used to measure the quality of candidate texts against reference texts for many different workflows. Though it is an n-gram based metric, it goes beyond traditional methods by factoring in elements such as precision, recall, and order to provide a comprehensive measure of text quality. For an in-depth justification behind METEOR's design choices, feel free to check out the [original paper](https://www.cs.cmu.edu/~alavie/METEOR/pdf/Banerjee-Lavie-2005-METEOR.pdf).
+
+## Implementation Details
+
+We define METEOR as the product of two components - the Unigram Precision/Recall Harmonic Mean, and Word Order Penalty. That is,
```
Comment (y27choi) [inline docs/metrics/meteor.md]: `Precision / Recall`

File: docs/metrics/meteor.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+??? info "FMean: Harmonic Mean of the Unigram Precision / Recall"
+    This is defined as
+
+    $$
+    FMean = \frac{10PR}{R + 9P}
+    $$
+
+    where P represents the unigram precision, and R represents the unigram recall. Here's a recap of [precision](precision.md) and [recall](recall.md). Notice that most of the weight is placed on the Recall component by design - this allows METEOR to prioritize the coverage of essential keywords in the candidate text.
```
Comment (y27choi) [inline docs/metrics/meteor.md]: `Recall` -> `recall`

File: docs/metrics/meteor.md
Comment (y27choi) [inline docs/metrics/meteor.md:1]: Use – instead

File: docs/metrics/meteor.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+**Candidate**: Joe likes to eat apples for lunch. <br>
+**Reference**: For lunch, joe likes to eat apples. <br>
+
+??? example "Step 1: Calculate FMean"
+    Our first step is trivial. Since both sentences contain the same words, our unigram precision and recall are both 1.0
+    As a result, our FMean is $\frac{10 \times 1.0 \times 1.0}{1.0 + 9 \times 1.0} = 1$
+
+??? example "Step 2: Calculate Word Order Penalty"
+    We can break up our candidate sentence into two chunks to map it to our refernece sentence.
```
Comment (y27choi) [inline docs/metrics/meteor.md]: typo: `reference`

File: docs/metrics/meteor.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    Between the two chunks, we have matched 7 unigrams. This gives us a penalty score of $0.5 \times \frac{7}{7} = 0.5$.
+
+??? example "Step 3: Calculate METEOR"
+    With our Penalty and FMean calculated, we can proceed with calculating the METEOR score.
+
+    $$
+    \text{METEOR} = 1 * (1 - 0.5) = 0.5.
+    $$
+
+    Despite having all the keywords of the reference sentence, our candidate had the wrong order and meaning! This is a massive improvement over something like ROUGE-1 which would not have considered the orders of the sentences, and given a perfect score of 1.0.
```
Comment (y27choi) [inline docs/metrics/meteor.md:94]: Very good example to showcase the advantage of using METEOR over ROUGE

File: docs/metrics/meteor.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+2. METEOR can fail on context. If we have two sentences "I am a big fan of Taylor Swift" (Reference) and "Fan of Taylor Swift I am big" (Candidate), METEOR would yield a good score. However, the candidate sentence makes little sense and intuitively shouldn't be given a good score. This is a limitation with all n-gram metrics, and not specific to METEOR.
+
+Limitations aside, METEOR is still a great metric to include in NLP workflows for measuring text similarity. Like other n-gram metrics, it is easy to compute and doesn't require extra hardware for inference. Furthermore, it is a noticible improvement over BLEU, and even ROUGE, in many ways - it places weight on *both* precision and recall, factors in word order, and generally does a better job at filtering out bad candidate texts, as seen in example 2. It is also a better judge of global coherence than BLEU and ROUGE, since it greedily looks for the largest chunks to calculate its penalty factor, rather than using a sliding context window of n-grams. METEOR is a powerful metric, and should be included in every NLP toolbox to give a holistic view of model performance.
```
Comment (y27choi) [inline docs/metrics/meteor.md]: `noticible` -> `noticeable`

**Review by y27choi — COMMENTED** [review]
Left some comments but overall 👍.
@marko-polo-cheno we should consider adding METEOR to our text summarization task.

**Comment by marko-polo-cheno** [comment]
> Left some comments but overall 👍. @marko-polo-cheno we should consider adding METEOR to our text summarization task.

Used to have METEOR while developing/testing, but I found it to be similar to BERT_F1 for the stratifications made, so I omitted it. In this case, the extra complexity (easy example + run-time) might not be worth it.
[link](https://trunk.kolena.io/yoohee/results?modelIds=631&workflow=Text+Summarization&testSuiteId=1192&evaluators=N4IgjGAcCcIFygGwGYzwCxgEwF8A04Y0W8SqG2%2Bh06pIKacmuB2ADG3QxbjkA)

File: docs/metrics/meteor.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+    $$
+    Penalty = 0.5 \times \frac{\text{# of Chunks}}{\text{# of Unigrams Matched}}
+    $$
+
+    For example, if our candidate sentence was "the president spoke to the audience" and our reference sentence was "the president *then* spoke to the audience", there would be two chunks – "the president" and "spoke to the audience" – and 6 unigrams matched. Notice that as the chunks decrease, so does the penalty, which results in a higher METEOR score. This is quite intuitive as a lower number of chunks translates to an enforced order and better alignment with the reference text.
+
+## Examples
+
+**Candidate**: Joe likes to eat apples for lunch. <br>
```
Comment (y27choi) [inline docs/metrics/meteor.md]: Sorry for nitpicking @billbai0102 but can you change these sentences in a similar format as BLEU or ROUGE docs?

Comment (billbai0102) [reply docs/metrics/meteor.md]: Example has been changed

File: docs/metrics/meteor.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+??? example "Step 3: Calculate METEOR"
+    With our Penalty and FMean calculated, we can proceed with calculating the METEOR score.
+
+    $$
+    \text{METEOR} = 1 * (1 - 0.125) = 0.875.
+    $$
+
+    Not bad! We have a pretty high score for two sentences that are semantically the same but have different orders.
+
+Lets try the same reference example with a slightly different candidate.
```
Comment (y27choi) [inline docs/metrics/meteor.md:68]: `Lets` -> `Let's`

**Comment by y27choi** [comment]
Let's add a link to BLEU / ROUGE docs now they are merged.

**Review by y27choi — APPROVED** [review]
> Let's add a link to BLEU doc now that's merged.

If you rather do this as another PR at the end of merging everything, that's fine with me too.

Left a minor comment that needs to be addressed but otherwise everything looks good.

File: docs/metrics/meteor.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    $$
+    \text{METEOR} = 1 * (1 - 0.5) = 0.5.
+    $$
+
+    Despite having all the keywords of the reference sentence, our candidate had the wrong order and meaning! This is a massive improvement over something like ROUGE-1 which would not have considered the orders of the sentences, and given a perfect score of 1.0.
+
+## Limitations and Advantages
+Although METEOR was created to address some of the major limitations of BLEU, it still comes with its own limitations.
+
+1. METEOR does not consider synonyms. Unlike embeddings-based metrics like BERT, it does not have a mechanism to quantify the similarity of words within the candidate and reference sentences. Thus, having two sentences like "She looked extremely happy at the surprise party." and "She appeared exceptionally joyful during the unexpected celebration." would yield a subobtimal score despite being very similar in meaning. That being said, METEOR has shown to have a higher correlation with human judgement than both BLEU and ROUGE, making it *generally* better than the two.
```
Comment (y27choi) [inline docs/metrics/meteor.md]: Should `BERT` be `BERTScore`?

**Pull request merged**